packages/utils/src/tracing.ts

```diff
@@ -36,3 +39,81 @@ export function extractTraceparentData(traceparent: string): TraceparentData | u
     parentSpanId: matches[2],
   };
 }
+
+export type PolymorphicRequestHeaders =
+  | Record<string, string | undefined>
+  | Array<[string, string]>
+  // the below is not preicsely the Header type used in Request, but it'll pass duck-typing
+  | {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      [key: string]: any;
+      append: (key: string, value: string) => void;
+      get: (key: string) => string;
+    };
+
+/**
+ * Add `sentry-trace` and `baggage` headers to a given fetch request
+ */
+export function addTracingHeadersToFetchRequest(
+  request: string | Request,
```

Comment:
`Request` is browser types - will break in Node fyi.

We need to move this to browser for now then.

Reply:
Ah wait nvm, it doesn't! We use it in another places just fine.

Reply:
No I think you're right, Node E2E tests are failing because of `Request`: https://github.com/getsentry/sentry-javascript/actions/runs/4597451953/jobs/8121008921?pr=7626#step:6:6075

Reply:
Will move it to tracing internal